relayer/src/link.rs

```diff
@@ -391,10 +397,39 @@ impl RelayPath {
         self.timeout_msgs = vec![];
     }
 
-    fn relay_from_events(&mut self) -> Result<(), LinkError> {
+    fn relay_pending_packets(&mut self) -> Result<(), LinkError> {
+        println!("clearing old packets on {:#?}", self.channel);
+        for _i in 0..MAX_ITER {
```

Comment:
More general question, which does not be addressed in this PR: would it be sensible to introduce a delay or even implement some kind of exponential backoff strategy when retrying an operation which has failed (in this case, relaying the pending packets)?

Reply:
Yeh, we've been avoiding the whole error handling for some time. We have this pattern **everywhere**! We blindly retry in a loop for all errors. We need to sit down and see how to deal with each error. We have RPC, light client, on-chain and internal errors. The most complex ones are the on-chain errors that are a bit more subtle (leaving aside errors in the relayer code) and also they are not typed (we just get an error message and the reason is embedded in a string). The retry with exponential backoff may apply for some cases, in some cases we should not retry (e.g. a transaction has already been submitted by another relayer), in other cases we should rebuild stuff, etc. I think we should start a little doc and try to cover all errors and what actions we should take. 
Because the problem is so broad I would leave this handling here for later.

Reply:
Opened https://github.com/informalsystems/ibc-rs/issues/712 for error handling.